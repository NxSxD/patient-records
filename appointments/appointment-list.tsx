import React from "react";
import styled from "styled-components";
import { Appointment } from "./types";
import dayjs from "dayjs";
import {
  ButtonPrimary,
  ListHeader,
  ListItemContainer,
  ListContainer,
  Empty,
  InfoLabel,
  ItemInfo,
} from "../primitives";
import Link from "next/link";

interface AppointmentListProps {
  appointments: Appointment[];
  onAddAppointment?: () => void;
}

export const AppointmentList: React.FC<AppointmentListProps> = ({
  appointments,
  onAddAppointment,
}) => {
  const onAdd = (ev: React.MouseEvent<HTMLButtonElement>) => {
    ev.preventDefault();
    ev.stopPropagation();
    onAddAppointment && onAddAppointment();
  };

  return (
    <ListContainer>
      <ListHeader>
        <span>Last 3 appointments</span>
        <ButtonPrimary onClick={onAdd}>&#43; Appointment</ButtonPrimary>
      </ListHeader>
      {appointments?.length > 0 ? (
        appointments.map((ap) => (
          <Link key={ap.id} href={`/appointments/${ap.id}`}>
            <AppointmentDisplay>
              <ItemInfo>
                <InfoLabel>Doctor</InfoLabel>
                <span>{ap.doctor}</span>
              </ItemInfo>
              <ItemInfo>
                <InfoLabel>Appointment Time</InfoLabel>
                <span>
                  {dayjs(ap.appointment_time).format("MM/DD/YY -- HH:MMa")}
                </span>
              </ItemInfo>
            </AppointmentDisplay>
          </Link>
        ))
      ) : (
        <Empty>Seems like you don&apos;t have any appointments yet.</Empty>
      )}
    </ListContainer>
  );
};

const AppointmentDisplay = styled(ListItemContainer)`
  justify-content: space-between;
  align-items: center;
`;
