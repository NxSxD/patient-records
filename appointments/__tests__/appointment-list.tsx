import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import { AppointmentList } from "../appointment-list";

import appointmentsJson from "./appointments.json";

describe("Appointment List", () => {
  const onAddAppointment = jest.fn();

  beforeEach(() => {
    onAddAppointment.mockReset();
  });

  it("should render according to snapshot", () => {
    const { container } = render(
      <AppointmentList
        appointments={appointmentsJson}
        onAddAppointment={onAddAppointment}
      />
    );

    expect(container).toMatchSnapshot();
  });

  it("should invoke onAddAppointment", () => {
    render(
      <AppointmentList
        appointments={appointmentsJson}
        onAddAppointment={onAddAppointment}
      />
    );

    fireEvent.click(screen.getByText(/\+ Appointment/));

    expect(onAddAppointment).toHaveBeenCalled();
  });
});
