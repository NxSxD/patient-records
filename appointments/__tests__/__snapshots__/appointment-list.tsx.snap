// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Appointment List should render according to snapshot 1`] = `
.c2 {
  height: 45px;
  padding: 0 1rem;
  font-size: 1.2rem;
  border: none;
  border-radius: 4px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-transition: background-color 250ms;
  transition: background-color 250ms;
  min-width: 120px;
  background-color: var(--primary-color);
  color: #fff;
}

.c2:hover {
  cursor: pointer;
  background-color: var(--primary-alt-color);
}

.c2:focus {
  outline: none;
}

.c0 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  height: 65px;
  border-radius: 4px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c5 {
  font-size: 0.9rem;
  margin-bottom: 0.5rem;
  color: var(--text-label-color);
}

.c4 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c3 {
  height: 65px;
  border-radius: 4px;
  border: 1px solid #ccc;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  padding: 0 1rem;
  margin-bottom: 1rem;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3:hover {
  cursor: pointer;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <span>
        Last 3 appointments
      </span>
      <button
        class="c2"
      >
        + Appointment
      </button>
    </div>
    <div
      class="c3"
    >
      <div
        class="c4"
      >
        <span
          class="c5"
        >
          Doctor
        </span>
        <span>
          Docror 1
        </span>
      </div>
      <div
        class="c4"
      >
        <span
          class="c5"
        >
          Appointment Time
        </span>
        <span>
          08/20/20 -- 17:08pm
        </span>
      </div>
    </div>
    <div
      class="c3"
    >
      <div
        class="c4"
      >
        <span
          class="c5"
        >
          Doctor
        </span>
        <span>
          Doctor 2
        </span>
      </div>
      <div
        class="c4"
      >
        <span
          class="c5"
        >
          Appointment Time
        </span>
        <span>
          08/21/20 -- 17:08pm
        </span>
      </div>
    </div>
    <div
      class="c3"
    >
      <div
        class="c4"
      >
        <span
          class="c5"
        >
          Doctor
        </span>
        <span>
          Doctor 3
        </span>
      </div>
      <div
        class="c4"
      >
        <span
          class="c5"
        >
          Appointment Time
        </span>
        <span>
          08/22/20 -- 17:08pm
        </span>
      </div>
    </div>
  </div>
</div>
`;
